activerecord/test/cases/log_subscriber_test.rb

```diff
@@ -177,11 +177,24 @@ def test_vebose_query_logs
 
     logger = TestDebugLogSubscriber.new
     logger.sql(Event.new(0, sql: "hi mom!"))
+    assert_equal 2, @logger.logged(:debug).size
```

Comment:
Could anyone pass this test locally?

```
% bin/test test/cases/log_subscriber_test.rb
Using sqlite3
Run options: --seed 3338

# Running:

.F

Failure:
LogSubscriberTest#test_vebose_query_logs [/Users/kamipo/src/github.com/rails/rails/activerecord/test/cases/log_subscriber_test.rb:180]:
Expected: 2
  Actual: 1


bin/test test/cases/log_subscriber_test.rb:175

.................

Finished in 0.062077s, 306.0739 runs/s, 2416.3726 assertions/s.
```

Reply:
Yes:

```
kaspers-macbook-pro:activerecord kasperhansen$ bin/test test/cases/log_subscriber_test.rb 
Using sqlite3
Run options: --seed 20837

# Running:

..................

Finished in 0.042548s, 423.0516 runs/s, 3478.4244 assertions/s.

18 runs, 148 assertions, 0 failures, 0 errors, 0 skips
```

Reply:
Wired... all `line.absolute_path` would matches to `PATHS_TO_IGNORE` then `extract_query_source_location` returns no location.

```diff
diff --git a/activerecord/lib/active_record/log_subscriber.rb b/activerecord/lib/active_record/log_subscriber.rb
index 1ae6840921..d44a32f39a 100644
--- a/activerecord/lib/active_record/log_subscriber.rb
+++ b/activerecord/lib/active_record/log_subscriber.rb
@@ -114,7 +114,10 @@ def log_query_source
       PATHS_TO_IGNORE = /\A(#{RAILS_GEM_ROOT}|#{RbConfig::CONFIG["rubylibdir"]})/
 
       def extract_query_source_location(locations)
-        locations.find { |line| line.absolute_path && !line.absolute_path.match?(PATHS_TO_IGNORE) }
+        p [:regex, PATHS_TO_IGNORE]
+        locations.find { |line|
+          p [line.absolute_path, !line.absolute_path.match?(PATHS_TO_IGNORE)]
+          line.absolute_path && !line.absolute_path.match?(PATHS_TO_IGNORE) }
       end
   end
 end
```

```
% bin/test test/cases/log_subscriber_test.rb -n test_vebose_query_logs
Using sqlite3
Run options: -n test_vebose_query_logs --seed 38744

# Running:

[:regex, /\A(\/Users\/kamipo\/src\/github.com\/rails\/rails\/|\/Users\/kamipo\/.rbenv\/versions\/2.5.1\/lib\/ruby\/2.5.0)/]
[false, "/Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/log_subscriber.rb"]
[false, "/Users/kamipo/src/github.com/rails/rails/activerecord/test/cases/log_subscriber_test.rb"]
[false, "/Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/log_subscriber.rb"]
[false, "/Users/kamipo/src/github.com/rails/rails/activerecord/test/cases/log_subscriber_test.rb"]
[false, "/Users/kamipo/src/github.com/rails/rails/vendor/bundle/ruby/2.5.0/gems/minitest-5.11.3/lib/minitest/test.rb"]
[false, "/Users/kamipo/src/github.com/rails/rails/vendor/bundle/ruby/2.5.0/gems/minitest-5.11.3/lib/minitest/test.rb"]
[false, "/Users/kamipo/src/github.com/rails/rails/vendor/bundle/ruby/2.5.0/gems/minitest-5.11.3/lib/minitest/test.rb"]
[false, "/Users/kamipo/src/github.com/rails/rails/vendor/bundle/ruby/2.5.0/gems/minitest-5.11.3/lib/minitest.rb"]
[false, "/Users/kamipo/src/github.com/rails/rails/vendor/bundle/ruby/2.5.0/gems/minitest-5.11.3/lib/minitest/test.rb"]
[false, "/Users/kamipo/src/github.com/rails/rails/vendor/bundle/ruby/2.5.0/gems/minitest-5.11.3/lib/minitest.rb"]
[false, "/Users/kamipo/src/github.com/rails/rails/vendor/bundle/ruby/2.5.0/gems/minitest-5.11.3/lib/minitest/test.rb"]
[false, "/Users/kamipo/src/github.com/rails/rails/vendor/bundle/ruby/2.5.0/gems/minitest-5.11.3/lib/minitest/test.rb"]
[false, "/Users/kamipo/src/github.com/rails/rails/vendor/bundle/ruby/2.5.0/gems/minitest-5.11.3/lib/minitest.rb"]
[false, "/Users/kamipo/src/github.com/rails/rails/vendor/bundle/ruby/2.5.0/gems/minitest-5.11.3/lib/minitest.rb"]
[false, "/Users/kamipo/src/github.com/rails/rails/vendor/bundle/ruby/2.5.0/gems/minitest-5.11.3/lib/minitest.rb"]
[false, "/Users/kamipo/src/github.com/rails/rails/vendor/bundle/ruby/2.5.0/gems/minitest-5.11.3/lib/minitest.rb"]
[false, "/Users/kamipo/src/github.com/rails/rails/vendor/bundle/ruby/2.5.0/gems/minitest-5.11.3/lib/minitest.rb"]
[false, "/Users/kamipo/src/github.com/rails/rails/vendor/bundle/ruby/2.5.0/gems/minitest-5.11.3/lib/minitest.rb"]
[false, "/Users/kamipo/src/github.com/rails/rails/vendor/bundle/ruby/2.5.0/gems/minitest-5.11.3/lib/minitest.rb"]
[false, "/Users/kamipo/src/github.com/rails/rails/vendor/bundle/ruby/2.5.0/gems/minitest-5.11.3/lib/minitest.rb"]
[false, "/Users/kamipo/src/github.com/rails/rails/railties/lib/rails/test_unit/line_filtering.rb"]
[false, "/Users/kamipo/src/github.com/rails/rails/vendor/bundle/ruby/2.5.0/gems/minitest-5.11.3/lib/minitest.rb"]
[false, "/Users/kamipo/src/github.com/rails/rails/vendor/bundle/ruby/2.5.0/gems/minitest-5.11.3/lib/minitest.rb"]
[false, "/Users/kamipo/src/github.com/rails/rails/vendor/bundle/ruby/2.5.0/gems/minitest-5.11.3/lib/minitest.rb"]
[false, "/Users/kamipo/src/github.com/rails/rails/vendor/bundle/ruby/2.5.0/gems/minitest-5.11.3/lib/minitest.rb"]
[false, "/Users/kamipo/src/github.com/rails/rails/vendor/bundle/ruby/2.5.0/gems/minitest-5.11.3/lib/minitest.rb"]
F

Failure:
LogSubscriberTest#test_vebose_query_logs [/Users/kamipo/src/github.com/rails/rails/activerecord/test/cases/log_subscriber_test.rb:180]:
Expected: 2
  Actual: 1


bin/test test/cases/log_subscriber_test.rb:175



Finished in 0.017528s, 57.0528 runs/s, 57.0528 assertions/s.
1 runs, 1 assertions, 1 failures, 0 errors, 0 skips
```

Reply:
@kamipo I think that the reason the tests are failing for you is that your gems that are installed by bundler are installed under `vendor/bundle` making everything fall inside the `RAILS_GEM_ROOT`.

For my install, my `minitest` in installed under `~/.rbenv` so it sits outside of the  `RAILS_GEM_ROOT` and it gets reported as the caller.

I think that the fix would be to change the ignored paths to match active record and active support lib directories (which will just return instrumentation code as the caller, so we would never want these) and the path to where the gems are installed by bundler. This would leave the test cases outside if the ignored paths, so the line in the test should be returned as the caller.

Reply:
@lsylvester definitely worth exploring! But we'd want to catch Active Storage queries as well, right?